### PR TITLE
8317831: compiler/codecache/CheckLargePages.java fails on OL 8.8 with unexpected memory string

### PR DESCRIPTION
**Pull request by yftsai**
Test CheckLargePages was broken by the previous changes:

[JDK-8310233](https://bugs.openjdk.org/browse/JDK-8310233) changes the pagesize logs from
```
Usable page sizes: 4k, 1G
```
to
```
Large page support enabled. Usable page sizes: 4k, 1G. Default large page size: 1G.
```

[JDK-8261894](https://bugs.openjdk.org/browse/JDK-8261894) removes `UseHugeTLBFS`. It was also removed from `os::can_execute_large_page_memory`, and `CodeCache::page_size` cannot  use huge pages anymore.

This change includes:
- The regular expression in CheckLargePages is updated to capture only the page sizes.
- The static huge page will be fixed by  [JDK-8319795](https://bugs.openjdk.org/browse/JDK-8319795).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8317831](https://bugs.openjdk.org/browse/JDK-8317831): compiler/codecache/CheckLargePages.java fails on OL 8.8 with unexpected memory string (**Bug** - P4)

### Reviewers
 * [Evgeny Astigeevich](https://openjdk.org/census#eastigeevich) (@eastig - Committer)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16962/head:pull/16962` \
`$ git checkout pull/16962`

Update a local copy of the PR: \
`$ git checkout pull/16962` \
`$ git pull https://git.openjdk.org/jdk.git pull/16962/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16962`

View PR using the GUI difftool: \
`$ git pr show -t 16962`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16962.diff">https://git.openjdk.org/jdk/pull/16962.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16962#issuecomment-1839833285)

### CODE REVIEW

File: src/hotspot/os/linux/os_linux.cpp [hunk 1]
```diff
@@ -4004,7 +4004,7 @@ bool os::can_commit_large_page_memory() {
 }
 
 bool os::can_execute_large_page_memory() {
-  return UseTransparentHugePages;
+  return UseLargePages;
```
Comment (eastig) [inline src/hotspot/os/linux/os_linux.cpp:4007]: Could you please exclude this change from the PR? The function will be removed by my PR.

In `test/hotspot/jtreg/ProblemList.txt` could you please change
```
compiler/codecache/CheckLargePages.java 8317831 linux-x64
```

to
```
compiler/codecache/CheckLargePages.java 8319795 linux-x64
```

**Review by eastig — CHANGES_REQUESTED** [review]

**Comment by eastig** [comment]
I tested my PR #16582 with the fixed test. The test passed as expected.

**Review by eastig — APPROVED** [review]
lgtm

**Review by dholmes-ora — APPROVED** [review]
I'm going to approve this test fix as the change to the regex itself seems reasonable, even though because the test remains excluded by [JDK-8319795](https://bugs.openjdk.org/browse/JDK-8319795) we won't actually be testing it. If any issues arise when the test is removed from the ProblemList it is expected that [JDK-8319795](https://bugs.openjdk.org/browse/JDK-8319795) will deal with that.

Thanks

**Pull request closed without merging**